docs/CustomPipelineGraphicsMod.md

```diff
@@ -0,0 +1,251 @@
+|``texmap``               | uint                             | v1    | The texture unit for this stage                                               |
+|``output_color``         | vec4                             | v1    | The final output color this stage produces                                    |
+
+
+`CustomShaderTevStageInputColor` is a single input TEV operation for a color value.  It has the following structure:
+
+|Name                     | Type | Since | Description                                     |
+|-------------------------|------|-------|-------------------------------------------------|
+|``input_type``           | uint | v1    | The input type of the input.  See details below |
+|``value``                | vec3 | v1    | The value of input                              |
+
+The `input_type` is defined as a `uint` but is effectively an enumeration.  it has the following values:
+
+|Name                                              | Since | Description                                                               |
+|--------------------------------------------------|-------|---------------------------------------------------------------------------|
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_PREV``       | v1    | The value is provided by the last stage                                   |
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_COLOR``      | v1    | The value is provided by the color data                                   |
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_TEX``        | v1    | The value is provided by a texture                                        |
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_RAS``        | v1    |                                                                           |
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_KONST``      | v1    | The value is a constant value defined by the software                     |
+|``CUSTOM_SHADER_TEV_STAGE_INPUT_TYPE_NUMERIC``    | v1    | The value is a constant numeric value like vec3(0, 0, 0) or vec3(1, 1, 1) |
+
+`CustomShaderTevStageInputAlpha` is a single input TEV operation for an alpha value.  It has the following structure:
+
+|Name                     | Type | Since | Description                                                                   |
+|-------------------------|------|-------|-------------------------------------------------------------------------------|
+|``input_type``           | uint | v1    | The input type of the input.  See `input_type` for color input stages         |
+|``value``                | uint | v1    | The value of input                                                            |
+
+
+## Examples
+
+Below are a handful of examples.
+
+### Single color
+
+The following shader displays the color red on the screen:
+
+```
+vec4 custom_main( in CustomShaderData data )
+{
+	return vec4(1.0, 0.0, 0.0, 1.0);
+}
+```
```

Comment:
Currently this renders as:

```
vec4 custom_main( in CustomShaderData data )
{
	return vec4(1.0, 0.0, 0.0, 1.0);
}
```

If you write

<pre lang="markdown">
```glsl
</pre>

then you get syntax highlighting, like this:

```glsl
vec4 custom_main( in CustomShaderData data )
{
	return vec4(1.0, 0.0, 0.0, 1.0);
}
```

(there's also <code>```json</code> and a bunch of other languages.)

Reply:
Appreciate it, I will update those!